android/java/org/chromium/chrome/browser/settings/BraveSearchEngineUtils.java

```diff
@@ -54,27 +81,26 @@ static private void initializeDSEPrefs() {
         }
     }
 
-    static private void doInitializeBraveSearchEngineStates() {
-        assert TemplateUrlServiceFactory.get().isLoaded();
+    static private void doInitializeBraveSearchEngineStates(Profile profile) {
+        assert BraveTemplateUrlServiceFactory.get(profile).isLoaded();
 
-        initializeDSEPrefs();
-        // Initially set standard dse as an active DSE.
-        updateActiveDSE(false);
+        initializeDSEPrefs(profile);
+        updateActiveDSE(profile);
     }
 
-    static public void setDSEPrefs(TemplateUrl templateUrl, boolean isPrivate) {
-        BraveSearchEngineAdapter.setDSEPrefs(templateUrl, isPrivate);
+    static public void setDSEPrefs(TemplateUrl templateUrl, Profile profile) {
+        BraveSearchEngineAdapter.setDSEPrefs(templateUrl, profile);
     }
 
-    static public void updateActiveDSE(boolean isPrivate) {
-        BraveSearchEngineAdapter.updateActiveDSE(isPrivate);
+    static public void updateActiveDSE(Profile profile) {
```

Comment:
Below is not a notice to the PR itself, but to the problem in general.

`updateActiveDSE` is still a fragile thing.
Maybe we will be able to get rid of it in the future, as when we disaplay current SE in settings, we know which one do we want to get or set; and when we doing search, `AutocompleteControllerAndroid` does have the correct profile pointer for the OTR profile. And I cannot give a better way for `TemplateUrlServiceFactory` patching.

Reply:
Yeah I agree, just that it will require changing everywhere using the service probably which is too big a change for now, maybe in the future we can go back and do a better refactor.